File: trek-management-frontend/src/services/wishlistService.ts
import apiClient from '@/api/apiClient';
import type { TrekSummaryResponse } from '@/types/api';

const API_URL = '/wishlist';

export const wishlistService = {
  getMyWishlist: async (): Promise<TrekSummaryResponse[]> => {
    const response = await apiClient.get(API_URL);
    return response.data;
  },

  getWishlistTrekIds: async (): Promise<string[]> => {
    const response = await apiClient.get(`${API_URL}/ids`);
    return response.data;
  },

  addToWishlist: async (trekId: string): Promise<void> => {
    await apiClient.post(`${API_URL}/${trekId}`);
  },

  removeFromWishlist: async (trekId: string): Promise<void> => {
    await apiClient.delete(`${API_URL}/${trekId}`);
  }
};

File: src/main/java/com/trekmanagement/wishlist/WishlistService.java
package com.trekmanagement.wishlist;

import com.trekmanagement.trek.dto.TrekSummaryResponse;
import java.util.List;
import java.util.UUID;

public interface WishlistService {
    void addToWishlist(UUID trekId);
    void removeFromWishlist(UUID trekId);
    List<TrekSummaryResponse> getMyWishlist();
    List<UUID> getWishlistTrekIds();
}
